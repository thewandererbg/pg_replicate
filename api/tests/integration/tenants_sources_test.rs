use crate::{
    common::test_app::{
        spawn_test_app, CreateTenantSourceRequest, CreateTenantSourceResponse, SourceResponse,
        TenantResponse,
    },
    integration::sources_test::{new_name, new_source_config},
};

#[tokio::test(flavor = "multi_thread")]
async fn tenant_and_source_can_be_created() {
    // Arrange
    let app = spawn_test_app().await;

    // Act
    let tenant_source = CreateTenantSourceRequest {
        tenant_id: "abcdefghijklmnopqrst".to_string(),
        tenant_name: "NewTenant".to_string(),
        source_name: new_name(),
        source_config: new_source_config(),
    };
    let response = app.create_tenant_source(&tenant_source).await;

    // Assert
    assert!(response.status().is_success());
    let response: CreateTenantSourceResponse = response
        .json()
        .await
        .expect("failed to deserialize response");
    assert_eq!(response.tenant_id, "abcdefghijklmnopqrst");
    assert_eq!(response.source_id, 1);

    let tenant_id = &response.tenant_id;
    let source_id = response.source_id;

    let response = app.read_tenant(tenant_id).await;
    let response: TenantResponse = response
        .json()
        .await
        .expect("failed to deserialize response");
    assert_eq!(&response.id, tenant_id);
    assert_eq!(response.name, tenant_source.tenant_name);

    let response = app.read_source(tenant_id, source_id).await;
    let response: SourceResponse = response
        .json()
        .await
        .expect("failed to deserialize response");
    assert_eq!(response.id, source_id);
    assert_eq!(&response.tenant_id, tenant_id);
    assert_eq!(response.name, tenant_source.source_name);
    insta::assert_debug_snapshot!(response.config);
}
