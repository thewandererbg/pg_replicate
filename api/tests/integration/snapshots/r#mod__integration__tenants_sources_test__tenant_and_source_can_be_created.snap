---
source: api/tests/integration/tenants_sources_test.rs
expression: response.config
---
SourceConfig {
    host: "localhost",
    port: 5432,
    name: "postgres",
    username: "postgres",
    password: Some(
        "postgres",
    ),
}
